src/scrabble.js

```diff
@@ -1,6 +1,98 @@
 class Scrabble {
-  // Write your implementation here
+  constructor (word) {
+    this.word = word
+  }
 
+  scoreLoop (string) {
+    let num = 0
+    for (let i = 0; i < string.length; i++) {
+      if (string[i].match(/[^A-Z{}[]]/i)) {
+        return 0
+      }
+      if (string[i].match(/[AEIOULNRST]/i)) {
+        num++
+      }
+      if (string[i].match(/[DG]/i)) {
+        num += 2
+      }
+      if (string[i].match(/[BCMP]/i)) {
+        num += 3
+      }
+      if (string[i].match(/[FHVWY]/i)) {
+        num += 4
+      }
+      if (string[i].match(/[K]/i)) {
+        num += 5
+      }
+      if (string[i].match(/[JX]/i)) {
+        num += 8
+      }
+      if (string[i].match(/[QZ]/i)) {
+        num += 10
+      }
+    }
+    return num
+  }
+
+  scoreMain () {
+    if (this.word == null) {
+      return 0
+    }
+
+    const wordTrimmed = this.word.trim()
+
+    return this.scoreLoop(wordTrimmed)
+  }
+
+  scoreDoubleTripleLetter (open, close) {
+    if (this.word == null) {
+      return 0
+    }
+
+    if (this.word.indexOf(open) === -1 && this.word.indexOf(close) === -1) {
+      return 0
+    }
+
+    const letterArray = []
+
+    for (let i = 0; i < this.word.length; i++) {
+      if (this.word[i + 2] !== undefined) {
```

Comment:
This is an interesting approach. What made you need to check if that position of the word was undefined?

Also, this might be a good usecase of a guard statement.
```js
for(let i = 0; i < this.word.length; i++) {
  if(this.word[i + 2] === undefined) {
     continue // this skips this iteration of the loop
  }
}
```

Reply:
In another problem, I had to check if array[i + 1] was undefined otherwise the code didn't work. However, here as I am checking if it is equal to a specific value, I realise now it isn't necessary as it is allowed to be undefined!!
I may have been using array[i+1].length which could have caused the problems before??